runtime-transaction/src/compute_budget_instruction_details.rs

```diff
@@ -44,10 +50,36 @@ impl ComputeBudgetInstructionDetails {
             }
         }
 
```

Comment:
Only when Compute Unite Limit is not requested to re-iterate instructions to collect builtin program counts. This bit of optimization is because `is_compute_budget_program()` is cheap, `get_program_kind()` on the other hand does additional HashMap lookup by Pubkey; Saving unnecessary calls to `get_program_kind()` perhaps justifies additional iteration.